### DISCUSSION TITLE
DFPlayer Mini, FN-M16P Error "SerialWrongStack" solved

### DISCUSSION THREAD

**Discussion by Programmierfreund** [arendst/Tasmota#11737] [Ideas]
I have bought a DFPlayer Mini clone "FN-M16P Embedded MP3 Audio Module" through Amazon (product name: [Innovateking-EU 2 Stücke DFPlayer Mini [MP3]), (https://www.amazon.de/Innovateking-EU-DFPlayer-Kompatibel-Unterst%C3%BCtzung-Laufwerk/dp/B07T89PFFD/). The actual microprocessor is a MH-ET Live MH2024K-24SS as can be seen well on the product photos at Amazon. Actually, the device should be compatible with the DFPlayer Mini, however, like some other buyers, I had the problem that not the whole command set worked. E.g. neither the adjustment of the volume, nor the direct selection of single MP3 files according to the naming scheme 0001.mp3, 0002.mp3 etc. worked. This made the device unusable for me. I couldn't get it to work with an Arduino Uno using the reference implementation of the library DFRobotDFPlayerMini.cpp as well as with the Tasmota driver xdrv_14_mp3.ino and an ESP8266. The commands "play first MP3 file" or "play next MP3 file" however worked in both cases.

Before I ordered another device now, I went deeper into the topic and took a closer look at the error message "SerialWrongStack: Get Wrong Stack" (error code: "7E FF 06 40 00 00 03 FE B8 EF"). In a very detailed documentation for the FN-M16P at Github you can find "Serial receiving error (a frame has not been received completely yet)"(https://github.com/DFRobot/DFRobotDFPlayerMini/blob/master/doc/FN-M16P%2BEmbedded%2BMP3%2BAudio%2BModule%2BDatasheet.pdf). Maybe it is a timing problem of the serial connection. It is strange that the problem occurs with an Arduino Uno as well as with an ESP8266. In the same documentation it is also mentioned that you can send the data packets without checksums. I tried this just "for fun" once and the complete command set works at once! I have now removed the checksum calculation from the Tasmota driver xdrv_14_mp3.ino, recompiled Tasmota with platform.io and everything works fine. Somebody should understand this.

Is the solution also relevant for others, so that I should open an issue if necessary?

**Comment by Jason2866** [comment]
It is not a issue it is a feature request for a not yet supported device. The right place is here.

**Reply by Programmierfreund** [reply]
Ok, thanks for your advice.

**Comment by barbudor** [comment]
May be a PR for a `#ifdef MP3_PLAYER_NO_CHECKSUM` to disable properly the checksum would do the trick ?

**Reply by Programmierfreund** [reply]
That sounds good - I would appreciate it.

Maybe someone with a similar problem with a FN-M16P could confirm that disabling the checksum is a suitable solution for him as well.
